script.js

```diff
@@ -0,0 +1,166 @@
+  }
+  allTasks.sort((a, b) => a.isCheck > b.isCheck ? 1 : a.isCheck < b.isCheck ? -1 : 0);
+
+  allTasks.map((item, index) => {
+    const container = document.createElement('div');
+    container.id = `task-${index}`;
+    container.className = 'task-container';
+
+    const checkbox = document.createElement('input');
+    checkbox.type = 'checkbox';
+    checkbox.checked = item.isCheck;
+    checkbox.className = 'task-checkbox';
+    checkbox.onchange = () => {
+      onChangeCheckbox(index);
+      render();
+    };
+    container.appendChild(checkbox);
+
+    if (index === activeEditTask) {
+      const inputTask = document.createElement('input');
+      inputTask.type = 'text';
+      inputTask.value = item.text;
+      inputTask.addEventListener('change', updateTaskText);
+      inputTask.addEventListener('blur', doneEditTask);
+      container.appendChild(inputTask);
+    } else {
+      const text = document.createElement('p');
+      text.innerText = item.text;
+      text.className = item.isCheck ? 'text-task done-text' : 'text-task';
+      container.appendChild(text);
+    }
+
+    if (!item.isCheck) {
+      if (index === activeEditTask) {
+        const imageDone = document.createElement('img');
+        imageDone.src = 'img/done.png';
+        imageDone.onclick = () => {
+          doneEditTask();
+        };
+        container.appendChild(imageDone);
+      } else {
+        const imageEdit = document.createElement('img');
+        imageEdit.src = 'img/edit.png';
+        imageEdit.onclick = () => {
+          activeEditTask = index;
+          render();
+        }
+        container.appendChild(imageEdit);
+      }
+    }
+    const imageDelete = document.createElement('img');
+    imageDelete.src = 'img/delete.png';
+    imageDelete.onclick = () => {
+      let itemIdDel = item.id;
+      deleteTask(index, itemIdDel);
+    }
+    container.appendChild(imageDelete);
+    content.appendChild(container);
+  });
+}
+
+const onChangeCheckbox = async (index) => {
+  let elArr = allTasks[index];
+  let { isCheck, text, id } = elArr;
+  isCheck = !isCheck;
+
+  const response = await fetch('http://localhost:8000/updateTask', {
+    method: 'PATCH',
+    headers: {
+      'Content-Type': 'application/json;charset=utf-8',
+      'Access-Control-Allow-Origin': '*'
+    },
+    body: JSON.stringify({
+      id: id,
+      text: text,
+      isCheck: isCheck
+    })
+  });
+  let result = await response.json();
+  allTasks = result.data;
+  render();
+}
+
+const deleteTask = async (index, itemIdDel) => {
+  const response = await fetch(`http://localhost:8000/deleteTask?id=${itemIdDel}`, {
+    method: 'DELETE',
+    headers: {
+      'Content-Type': 'application/json;charset=utf-8',
+      'Access-Control-Allow-Origin': '*'
+    },
+  });
+  let result = await response.json();
+  allTasks = result.data;
+  render();
+}
+
+const updateTaskText = async (event) => {
+  let newElArr = allTasks[activeEditTask];
+  let { id, text } = newElArr;
```

Comment:
why let?

Reply:
if I use const - TypeError: Assignment to constant variable.